src/common/Search/index.js

```diff
@@ -0,0 +1,33 @@
+import debounce from "lodash.debounce";
+import { useLocation } from "react-router-dom";
+import { StyledSearch } from "./styled";
+import searchQueryParamName from "./searchQueryParamName";
+import { useReplaceQueryParameter } from "./queryParameterHooks";
+
+export const Search = () => {
+  const location = useLocation();
+  const replaceQueryParameter = useReplaceQueryParameter();
+
+  const onInputChange = debounce(({ target }) => {
+    replaceQueryParameter({
+      key: searchQueryParamName,
+      value: target.value.trim() !== "" ? target.value : undefined,
+    });
+
+    target.value = "";
```

Comment:
I think, it is disturbing when the query disappears after you stop typing 😁

Reply:
You're right.